### PR TITLE
MS-621 Upgrades

### PR DESCRIPTION
**Pull request by luhmirin-s**
* Added regression test cases to ensure that API does not change during refactoring
* Update build tooling to latest versions
* Translated all files to Kotlin with small changes to make the implementation more "kotlin-like"

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -5,30 +5,50 @@ buildscript {
         maven { url "https://plugins.gradle.org/m2/" }
     }
     dependencies {
-        classpath "com.android.tools.build:gradle:7.4.0"
-        classpath "io.github.gradle-nexus:publish-plugin:1.3.0"
+        classpath "com.android.tools.build:gradle:8.3.2"
```
Comment (meladRaouf) [inline build.gradle]: Can you use the latest gradle version ?

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Tue Jan 24 14:55:27 CET 2023
+#Wed Aug 14 11:22:29 EEST 2024
 distributionBase=GRADLE_USER_HOME
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.0-all.zip
 distributionPath=wrapper/dists
-zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.4-bin.zip
```
Comment (meladRaouf) [inline gradle/wrapper/gradle-wrapper.properties]: Also, here the wrapper version 8.7

File: src/main/java/com/simprints/libsimprints/Registration.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import kotlinx.parcelize.Parcelize
+
+@SuppressWarnings("unused", "WeakerAccess")
+@Parcelize
+data class Registration @JvmOverloads constructor(
+    val guid: String,
+    private val templates: MutableMap<FingerIdentifier, ByteArray> = mutableMapOf()
+) : Parcelable {
+
+    fun setTemplate(fingerId: FingerIdentifier, fingerTemplate: ByteArray) {
```
Comment (meladRaouf) [inline src/main/java/com/simprints/libsimprints/Registration.kt:13]: What about face temeplates?

**Review by meladRaouf — APPROVED** [review]

**Review by alexandr-simprints — APPROVED** [review]

File: src/main/java/com/simprints/libsimprints/Registration.kt (hunk 3 above)
Comment (luhmirin-s) [reply src/main/java/com/simprints/libsimprints/Registration.kt:13]: Hmm, I just translated the code explicitly not changing the API.

I will add a subtask to add the face templates well for the "improvements" part

**Pull request merged**